src/main/java/org/example/homework5/Task3.java

```diff
@@ -0,0 +1,36 @@
+package org.example.homework5;
+
+import java.util.Random;
+
+/**
+ * Number Generator
+ */
+public class Task3 {
+    public static void main(String[] args) {
+        // Array to track occurrences of each number
+        int[] occurrences = new int[101];
+
+        int sum = 0;
+        int count = 0;
+
+        Random random = new Random();
+
+        while (true) {
+            int randomNumber = random.nextInt(100) + 1;
+
+            System.out.println("Generated number: " + randomNumber);
+
+            if (occurrences[randomNumber] == 1) {
```

Comment:
Previously we agreed that we process the number when it occurs for the second time, and only after it we quit the loop. Since the number is already generated, it should be added to the sum